lib/tasks/database_updates/create_unique_site_banner_details_for_each_channel.rake

```diff
@@ -0,0 +1,43 @@
+namespace :database_updates do
+  desc "Create a unique site banner detail row for each channel that shares one on the publisher level"
+  task create_unique_site_banner_details_for_each_channel: :environment do
+    could_not_update = []
+    channels = Channel.where.missing(:site_banner)
+    channel_count = channels.count
+    puts "#{channels.count} channels have no site banner associated."
+
+    channels.each do |channel|
+      # Create a new site_banner and copy details from the default_site_banner, if existing
+      default_site_banner = channel.publisher.site_banners.detect { |sb| sb.id == channel.publisher.default_site_banner_id }
+
+      if default_site_banner
+        new_site_banner = SiteBanner.new(default_site_banner.attributes.except("id"))
+        new_site_banner.channel_id = channel.id
+
+        # Copy the attached images (logo and background_image) if they exist
+        if default_site_banner.logo.attached?
+          new_site_banner.logo.attach(default_site_banner.logo.blob)
+        end
+        if default_site_banner.background_image.attached?
+          new_site_banner.background_image.attach(default_site_banner.background_image.blob)
```

Comment:
This is going to potentially do a lot of traffic to s3. Perhaps the core logic here can be extracted to a sidekiq job that we can launch async so all of that can be done independently and then this job can execute fairly quickly.